### DISCUSSION TITLE
Substitute function definitions in SymPy expressions

### DISCUSSION THREAD

**Discussion by vladislavstrmlt** [sympy/sympy#26575] [Q&A] [answered]
Let's say I have an expression, such as `sympy.core.relational.Equality` containing an "unknown" function, like so.
```python
import sympy as sp

f = sp.Function(sp.Symbol("f", real=True))
x, y = sp.symbols(("x","y"), real=True)

equation = sp.Eq(f(x+y), f(x)+y)  # The functional equation f(x+y) = f(x)+y
```
My question is: Is there any way to substitute this function with its definition, for example if I suddenly know that it's $f(x)=x+2.$ The below doesn't work, as it only replaces exactly `f(x)`, with `x+2` not `f(x+y)` with `x+y+2`
```python
equation.subs(f(x), x+2)
```
This returns `Eq(f(x+y), x+2 + y)`, rather than the expected `Eq(x+y+2, x+2 + y)` (which then evaluates to `sympy.logic.boolalg.BooleanTrue`).

Essentially, I am trying to make a "check" method to check if a given function actually satisfies the condition, where it for example would return True when given the above, since it satisfies it. (I know this is probably not possible _completely in general_ covering all edge cases by Richardson's theorem, but it's gotta be possible to do the above(?))

Any help, links or anything is appreciated 🌟

**Comment by oscarbenjamin** [comment, marked as answer]
You can do this with `replace`:
```python
In [7]: a = Wild('a')

In [8]: f(x+y).replace(f(a), a+2)
Out[8]: x + y + 2

In [9]: equation.replace(f(a), a + 2)
Out[9]: True
```
